Python/pyhash.c

```diff
@@ -442,13 +442,13 @@ siphash24(uint64_t k0, uint64_t k1, const void *src, Py_ssize_t src_sz) {
     t = 0;
     pt = (uint8_t *)&t;
     switch (src_sz) {
-        case 7: pt[6] = in[6]; /* fall through */
-        case 6: pt[5] = in[5]; /* fall through */
-        case 5: pt[4] = in[4]; /* fall through */
+        case 7: pt[6] = in[6]; _Py_FALLTHROUGH;
+        case 6: pt[5] = in[5]; _Py_FALLTHROUGH;
+        case 5: pt[4] = in[4]; _Py_FALLTHROUGH;
         case 4: memcpy(pt, in, sizeof(uint32_t)); break;
-        case 3: pt[2] = in[2]; /* fall through */
-        case 2: pt[1] = in[1]; /* fall through */
-        case 1: pt[0] = in[0]; /* fall through */
+        case 3: pt[2] = in[2]; _Py_FALLTHROUGH;
+        case 2: pt[1] = in[1]; _Py_FALLTHROUGH;
+        case 1: pt[0] = in[0]; break;
     }
```

Comment:
Could this be sorted out using `default`?

```suggestion
        case 2: pt[1] = in[1]; _Py_FALLTHROUGH;
        case 1: pt[0] = in[0]; _Py_FALLTHROUGH;
        default: /* Do nothing. */
    }
```

That would also make it more explicit.  (Also I thought there is a compiler warning for no having `default`.)  Then again, your change using `break` is probably fine given the numerical progression of the case values.

Reply:
Adding `default:` case would change the generated machine code and can make it less efficient. There is nothing to do for case 0 for example. This code is performance sensitive. `break;` is just a standard for switch/case.

Reply:
Also, I'm the one who added the `/* fall through */` comment by mistake in 2017 :-) https://github.com/python/cpython/commit/c0e77364ca29df6cfb311e79892955c92bd8e595